libs/entitlement/src/main/java/org/elasticsearch/entitlement/initialization/EntitlementInitialization.java

```diff
@@ -38,7 +54,45 @@ public static EntitlementChecker checker() {
 
     // Note: referenced by agent reflectively
     public static void initialize(Instrumentation inst) throws Exception {
-        manager = new ElasticsearchEntitlementChecker();
+        Map<String, Policy> pluginPolicies = new HashMap<>();
+
+        for (Map.Entry<Path, Boolean> entry : EntitlementBootstrap.pluginData.entrySet()) {
```

Comment:
nit: can this parsing/validation be moved to a helper method so initialize stays small and easy to understand?